pkg/install/condition.go

```diff
@@ -18,15 +17,9 @@ func (i *Installer) bootstrapConfigMapReady() (bool, error) {
 func (i *Installer) apiServersReady() (bool, error) {
 	apiserver, err := i.configcli.ConfigV1().ClusterOperators().Get("kube-apiserver", metav1.GetOptions{})
 	if err == nil {
-		m := make(map[configv1.ClusterStatusConditionType]configv1.ConditionStatus, len(apiserver.Status.Conditions))
-		for _, cond := range apiserver.Status.Conditions {
-			m[cond.Type] = cond.Status
-		}
-		if m[configv1.OperatorAvailable] == configv1.ConditionTrue && m[configv1.OperatorProgressing] == configv1.ConditionFalse {
-			return true, nil
-		}
+		return isOperatorAvailable(apiserver), nil
 	}
-	return false, nil
+	return false, err
```

Comment:
the reason this nil is here is for error tolerance - rather than giving up if we get an error, we retry until we time out.

Please refactor these functions into the form:
```
foo, err := ...Get()
if err != nil {
    // be tolerant (write a better comment please)
    return false, nil
}
return isOperatorAvailable(), nil
```

Reply:
I added a comment at the beginning of the file since it applies to all condition functions.
Also refactored the functions to the suggested format.